shell/common/platform_util_linux.cc

```diff
@@ -32,54 +49,61 @@ bool XDGUtil(const std::vector<std::string>& argv, const bool wait_for_exit) {
   if (!process.IsValid())
     return false;
 
-  if (wait_for_exit) {
+  if (wait_for_exit || !callback.is_null()) {
     int exit_code = -1;
     if (!process.WaitForExit(&exit_code))
```

Comment:
If the process doesn't wait for exit then the api's promise will be unhandled. The callback should be handled here.